server/src/main/java/com/objectcomputing/checkins/services/guild/GuildController.java

```diff
@@ -41,59 +56,95 @@ public GuildController(GuildServices guildService) {
 
-    @Post(value = "/")
-    public HttpResponse<Guild> createAGuild(@Body @Valid GuildCreateDTO guild, HttpRequest<GuildCreateDTO> request) {
-        Guild newGuild = guildService.save(new Guild(guild.getName(), guild.getDescription()));
-        return HttpResponse
-                .created(newGuild)
-                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getUri(), newGuild.getId()))));
     }
 
-    /**
-     * Load the current guilds into checkinsdb.
-     *
-     * @param guildsList, array of {@link GuildCreateDTO guild create dto} to load {@link Guild guild(s)}
-     */
-
-    @Post("/guilds")
-    public HttpResponse<?> loadGuilds(@Body @NotNull @Valid List<GuildCreateDTO> guildsList, HttpRequest<List<GuildCreateDTO>> request) {
-        List<String> errors = new ArrayList<>();
-        List<Guild> guildsCreated = new ArrayList<>();
-        for (GuildCreateDTO guildDTO : guildsList) {
-            Guild guild = new Guild(guildDTO.getName(), guildDTO.getDescription());
-            try {
-                guildService.save(guild);
-                guildsCreated.add(guild);
-            } catch (GuildBadArgException e) {
-                errors.add(String.format("Guild %s was not added because: %s", guild.getName(), e.getMessage()));
-            }
-        }
-        if (errors.isEmpty()) {
-            return HttpResponse.created(guildsCreated).headers(headers ->
-                    headers.location(request.getUri()));
-        } else {
-            return HttpResponse.badRequest(errors).headers(headers ->
-                    headers.location(request.getUri()));
-        }
-    }
+     /**
+      * Load the current guilds into checkinsdb.
+      *
+      * @param guildsList, array of {@link GuildCreateDTO guild create dto} to load {@link Guild guild(s)}
+      */
+     @Post("/guilds")
+     public Single<HttpResponse<?>> loadGuilds(@Body @NotNull @Valid List<GuildCreateDTO> guildsList, HttpRequest<List<GuildCreateDTO>> request) {
+
+         return Single.fromCallable(() -> {
+             List<String> errors = new ArrayList<>();
+             List<Guild> guildCreated = new ArrayList<>();
+             for (GuildCreateDTO guildDTO : guildsList) {
+                 Guild guild = new Guild(guildDTO.getName(),
+                         guildDTO.getDescription());
+                 try {
+                     guildService.save(guild);
+                     guildCreated.add(guild);
+                 } catch (CompositeException e) {
+                     errors.add(String.format("Guild %s was not added because: %s", guild.getName(),
+                             guild.getDescription(), e.getMessage()));
+                 }
+             }
+             if (errors.isEmpty()) {
+                 return guildCreated;
+             }
+             throw new GuildBulkLoadException(errors);
+         }).map(guildCreated -> HttpResponse.created(guildCreated)
```

Comment:
The single needs to observe on the main event loop and then the map needs to observe on the io loop. Otherwise we aren't necessarily using the right threads for this process.

Reply:
Based on chats with John and Michael we have decided to remove the bulk load end points and the associated tests, created an issue for it.